File: nginx/MASZ/src/app/models/QuickSearchEntry.ts
import { AutoModEventTableEntry } from "./AutoModEventTableEntry";
import { IModCaseTableEntry } from "./IModCaseTableEntry";

export interface QuickSearchEntry {
    entry: IModCaseTableEntry|AutoModEventTableEntry;
    createdAt: Date;
    quickSearchEntryType: number;
}

File: backend/Bot/Models/QuickSearchEntry.cs
using Bot.Abstractions;
using Bot.Enums;

namespace Bot.Models;

public class QuickSearchEntry<T> : IQuickSearchEntry
{
    public T Entry { get; set; }
    public QuickSearchEntryType QuickSearchEntryType { get; set; }
    public DateTime CreatedAt { get; set; }
}
